### ISSUE TITLE
Ubuntu 16.04 "plugin NeedSecrets request #1 failed"

### ISSUE THREAD

**Issue by siriuslee** [nm-l2tp/network-manager-l2tp#16]
This is possibly related to #15, though since it's on Ubuntu and not Gentoo, things may be quite different. Attempting to connect to a VPN only prompts for a password on the first connection attempt after a reboot. This attempt fails to connect and any subsequent attempts do not prompt for a password. Attached are logs taken from `sudo journalctl -b`.

[nm-l2tp-failed-connection.txt](https://github.com/nm-l2tp/network-manager-l2tp/files/392593/nm-l2tp-failed-connection.txt)

**Comment by dkosovic** [comment]
Sorry I don't have any suggestion yet, I'm still looking analysing your log file and thinking about it. I might have seen something similar on OpenSUSE with strongSwan

I can tell from the following line in the log :

    charon[6382]: 00[CFG]   loaded IKE secret for %any %any

which indicates you didn't fill in the Group Name and Gateway ID fields in the NetworkManager-l2tp IPsec options dialog box, so doesn't appear to be the same issue as the Gentoo Libreswan issue. 

You mentioned Antergos and AUR in the other issue. Just curious, on Antergos were you using Libreswan or Strongswan  ?

**Comment by dkosovic** [comment]
But both issues seem to raise a problem when the IPsec child processes go astray, there ought to be a timeout and the processes killed when they do.

**Comment by siriuslee** [comment]
Turns out that the problem was indeed strongSwan. On Antergos I was using Libreswan (I had been trying a lot of different packages and apparently moved to libreswan at some point). I just installed libreswan 3.15 using the .deb package found here: https://download.libreswan.org/binaries/ubuntu/vivid/all/. Now it works! Authorization is quick and everything seems to be functional. 

Thanks so much for your help and for the awesome package. It kept me from having to switch to a mac for work :). Do let me know if I can help in any way with debugging on Ubuntu.
